lightkurve/lightcurve.py

```diff
@@ -1602,7 +1602,13 @@ def to_periodogram(self, method="lombscargle", **kwargs):
             return BoxLeastSquaresPeriodogram.from_lightcurve(lc=self, **kwargs)
         else:
             from . import LombScarglePeriodogram
-            return LombScarglePeriodogram.from_lightcurve(lc=self, **kwargs)
+
+            if (~np.isfinite(self.flux)).any():
+                log.warning('Lightcurve contains NaN values.'
+                            'These are removed before creating the periodogram.')
```

Comment:
```suggestion
                log.info('Lightcurve contains NaN values.'
                            'These are removed before creating the periodogram.')
```
I would suggest not to remove the message entirely, going from previous behaviour of raising an exception to silently "correcting" the same input without any feedback to the user. See general discussion for alternatives.